console/console.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2022 Red Hat OpenShift Data Federation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package console
+
+import (
+	"strings"
+
+	consolev1alpha1 "github.com/openshift/api/console/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const MAIN_BASE_PATH = "/"
+const COMPATIBILITY_BASE_PATH = "/compatibility/"
+
+func GetDeployment(namespace string) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mcg-ms-console",
+			Namespace: namespace,
+		},
+	}
+}
+
+func GetService(port int, namespace string) *apiv1.Service {
+	return &apiv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mcg-ms-console-service",
+			Namespace: namespace,
+			Annotations: map[string]string{
+				//TODO: Replace "addon-mcg-osd-parameters" with appr. secret
+				"service.alpha.openshift.io/serving-cert-secret-name": "addon-mcg-osd-parameters",
+			},
+			Labels: map[string]string{
+				"app": "mcg-ms-console",
+			},
+		},
+		Spec: apiv1.ServiceSpec{
+			Ports: []apiv1.ServicePort{
+				{Protocol: "TCP",
+					TargetPort: intstr.IntOrString{IntVal: int32(port)},
+					Port:       int32(port),
+					Name:       "console-port",
+				},
+			},
+			Selector: map[string]string{
+				"app": "mcg-ms-console",
+			},
+			Type: "ClusterIP",
+		},
+	}
+}
+
+func GetConsolePluginCR(consolePort int, basePath string, serviceNamespace string) *consolev1alpha1.ConsolePlugin {
+	return &consolev1alpha1.ConsolePlugin{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "mcg-ms-console",
+		},
+		Spec: consolev1alpha1.ConsolePluginSpec{
+			DisplayName: "mcg Plugin",
+			Service: consolev1alpha1.ConsolePluginService{
+				Name:      "mcg-ms-console-service",
+				Namespace: serviceNamespace,
+				Port:      int32(consolePort),
+				BasePath:  basePath,
+			},
+		},
+	}
+}
+
+func GetBasePath(clusterVersion string) string {
+	if strings.Contains(clusterVersion, "4.11") {
```

Comment:
only for version 4.11 we need to give a specific path, is that right?

Reply:
yeah... but it is not required exactly right now... may be in future when we use OCP 4.11... then we need to make changes to mcg-ms-console as well...
for first release we are focusing on OCP 4.10... so we can either keep it or remove it whatever is alright...

Reply:
I think this is something we need to take a step back and think about. Do we include these changes right now (given that we do not need them at this very moment, but might in the future), or do we remove them and just merge the bare minimum that's required to reconcile UI plugin with the operator?

Reply:
> only for version 4.11 we need to give a specific path, is that right?

yes.

Reply:
> I think this is something we need to take a step back and think about. Do we include these changes right now (given that we do not need them at this very moment, but might in the future), or do we remove them and just merge the bare minimum that's required to reconcile UI plugin with the operator?


Just want to quote the basepath definition here:

```
// basePath is the path to the plugin's assets. The primary asset it the
// manifest file called `plugin-manifest.json`, which is a JSON document
// that contains metadata about the plugin and the extensions.
```

In versions < 4.11, the metadata file is stored under `/`, later to 4.11 specific path can be provided. It doesn't have any special reason to not be there as it points to a place holder and it is just a part of infra provided by the CR. So it is up to the pod whether it uses it or ignores it.

Reply:
`In versions < 4.11, the metadata file is stored under /, later to 4.11 specific path can be provided.` --> **this is wrong**
in 4.11 as well basepath can be `/`, basically it depends...
in ODF we have 3 branches example: `release-4.10` and `release-4.10-compatibility` and `release-4.11`
OCP 4.11 when uses ODF 4.11 --> basepath is `/` 
OCP 4.11 when uses ODF 4.10 --> basepath is `/compatibility`
OCP 4.10 when uses ODF 4.10 --> basepath is again `/`

`It doesn't have any special reason to not be there as it points to a place holder and it is just a part of infra provided by the CR. So it is up to the pod whether it uses it or ignores it.` --> I agree... it is up to you guys whether you want to add it or not... anything is fine... these are just few lines.... when we support mcg-ms-console with OCP 4.11 we will have to revisit it later...

Reply:
ACK, sounds good.